api/lib/domain/usecases/authenticate-user.js

```diff
@@ -53,7 +54,8 @@ module.exports = async function authenticateUser({
     );
 
     if (shouldChangePassword) {
-      throw new UserShouldChangePasswordError();
+      const passwordResetToken = tokenService.createPasswordResetToken(foundUser.id);
+      throw new UserShouldChangePasswordError(undefined, passwordResetToken);
```

Comment:
Au lieu de passer `undefined` il est possible en js de ne rien passer du tout je crois : 
```suggestion
      throw new UserShouldChangePasswordError(, passwordResetToken);
```

Reply:
Je pense que le plus simple serait d'utiliser un objet comme paramètre car actuellement le constructeur attends 2 paramètres dont le 1er qui a une valeur par défaut.

```suggestion
throw new UserShouldChangePasswordError({ passwordResetToken });
```

Reply:
@Anne-Gaelle-S : malheureusement, ça ne fonctionne pas. Tu pensais peut-être à la destructuration de tableau comme `const [, itemB] = itemsArray` ?

@igorissen Ça aurait été plus élégant mais je n'aime pas trop l'idée de changer la signature du constructeur alors qu'on étend un objet natif qui a cette signature (comme globalement toutes nos erreurs). Mais je reste ouvert au débat !

Reply:
@iwazaru comme tu le dis on étend une classe de base ce qui implique l'appel à `super(...)` si tu modifies la signature du constructeur ce qui a été fait avec `DomainError extends Error`. On pourrait aussi l'appliquer pour cette erreur `UserShouldChangePasswordError ` mais en utilisant un objet comme paramètre.

```js
class DomainError extends Error {
  constructor(message, code, meta) {
    super(message);
    this.code = code;
    this.meta = meta;
  }
}

class MySuperDomainError extends DomainError {}
const mySuperDomainError = new MySuperDomainError('message', 233, 'meta');

class SecondSuperDomainError extends DomainError {
  constructor({ message = 'default message', code, meta, extras }) {
    super(message, code, meta);
    this.extras = extras;
  }
}
const secondSuperDomainError = new SecondSuperDomainError({extras: {a:'a'}});
```